File: app/src/main/java/com/fieldservice/app/data/AppContainer.java
package com.fieldservice.app.data;

import com.fieldservice.app.data.repository.MockAuthRepository;
import com.fieldservice.app.data.repository.MockTicketRepository;
import com.fieldservice.app.domain.repository.AuthRepository;
import com.fieldservice.app.domain.repository.TicketRepository;

/**
 * Provedor manual das dependências do app (sem Hilt, para manter a primeira versão simples).
 * As telas dependem apenas das interfaces {@link AuthRepository}/{@link TicketRepository}; trocar
 * a implementação mock por uma baseada em API significa mudar apenas esta classe.
 */
public final class AppContainer {

    private static final AuthRepository authRepository = new MockAuthRepository();
    private static final TicketRepository ticketRepository = new MockTicketRepository();

    private AppContainer() {
    }

    public static AuthRepository getAuthRepository() {
        return authRepository;
    }

    public static TicketRepository getTicketRepository() {
        return ticketRepository;
    }
}

File: legacy-kotlin/app/src/main/java/com/fieldservice/app/data/AppContainer.kt
package com.fieldservice.app.data

import com.fieldservice.app.data.repository.MockAuthRepository
import com.fieldservice.app.data.repository.MockTicketRepository
import com.fieldservice.app.domain.repository.AuthRepository
import com.fieldservice.app.domain.repository.TicketRepository

/**
 * Provedor manual das dependências do app (sem Hilt, para manter a primeira versão simples).
 * As telas dependem apenas das interfaces [AuthRepository]/[TicketRepository]; trocar a
 * implementação mock por uma baseada em API significa mudar apenas este arquivo.
 */
object AppContainer {
    val authRepository: AuthRepository by lazy { MockAuthRepository() }
    val ticketRepository: TicketRepository by lazy { MockTicketRepository() }
}
